### ISSUE TITLE
Trying to open Intel Quartis Prime 18.1

### ISSUE THREAD

**Issue by JoelSerranoLozano** [FEX-Emu/FEX#2086]
Hello when I try to execute quartus using FEXBash I run into the following error:
`FEXBash-root@oriol-Parallels-ARM-Virtual-Machine:/home/oriol/Intel_Quartus/quartus/bin> ./quartus
quartus: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory
`
Thanks

**Comment by Sonicadvance1** [comment]
That looks like it is currently just missing libpng12, which Ubuntu 22.04 doesn't ship anymore, thus it isn't in the rootfs.
Maybe getting some libpng12 package installed inside the rootfs will fix this but will just need to be tried.

**Comment by Sonicadvance1** [comment]
I downloaded the latest 21.1 trial and that seemed to work for me. If you manage to get the older version's libraries installed then I see no reason why the old version wouldn't work.

![Screen Shot 2022-10-18 at 2 21 37 AM](https://user-images.githubusercontent.com/1018829/196391038-13e9ec5c-bc7e-420c-9d08-8903cddb5e31.png)

**Comment by phire** [comment]
Yeah... but with FPGA toolchains, people typically get stuck on an older version. The last version that supported their FPGA or something. 

> which Ubuntu 22.04 doesn't ship anymore

Did Ubuntu/Debian really drop the libpng12 package? That was quite a common requirement that a lot of binaries assumed would be there. Arch still ships it, and it would be worth shipping in our rootfs.

**Comment by Sonicadvance1** [comment]
Yep, it's been removed for quite a while now.

**Comment by JoelSerranoLozano** [comment]
Would be cool if you could add it I hace downloaded latest version anyway.
By the way is Apple Silicon supoort coming for asahi Linux? Thanks for your help!!

**Comment by Sonicadvance1** [comment]
Once Asahi Linux adds support for 4k page size kernels then FEX-Emu will work on it.
The hardware specific features of Apple Silicon will require their team to implement kernel support and expose it to userspace.

**Comment by Sonicadvance1** [comment]
Not a FEX bug, just a legacy application breaking due to packages being removed, not our fault.
